### ISSUE TITLE
Camera uploads should not run on metered networks

### ISSUE THREAD

**Issue by agross** [nextcloud/android#6851]
### Steps to reproduce
1. Connect to a metered WiFi network
2. Set up auto uploads
3. Take some pictures to trigger auto upload

### Expected behaviour

Uploads should be deferred until a non-metered network connection is available.

### Actual behaviour

Uploads use up the metered network's data volume.

### Can you reproduce this problem on https://try.nextcloud.com?

Server side does not matter.

### Environment data
Android version: 9

Device model: Poco F1

Stock or customized system: Stock

Nextcloud app version: Latest

Nextcloud server version: 19.0.2

Reverse proxy: nginx

**Comment by tobiasKaminsky** [comment]
There is the option "only upload on unmetered wi-fi". 
Is this checked?
If yes, we rely on Androids capability to inform apps if the active network is metered.
Can you check if Android correctly recognizes this?

**Comment by agross** [comment]
Yes, all my synced directories have the "only unmetered" option checked and Android also displays the network as being metered.

**Comment by agross** [comment]
Thank you, @tobiasKaminsky!

**Comment by agross** [comment]
This broke again. How can the issue be reopened?

**Comment by agross** [comment]
Would someone please reopen this issue?

**Comment by nottheend** [comment]
Does it make sense to remove the milestone from this issue and re-assign it to a future one?
This issue might be overlooked otherwise. Thanks

**Comment by nottheend** [comment]
Is there a possibility to implement a switch to disable all automated uploads?

If massive uploads happen on metered wifis, the connection could be rejected by the provider for the complete phone

**Comment by joshtrichards** [comment]
> Is there a possibility to implement a switch to disable all automated uploads?

@nottheend There's an existing issue for that #383

**Comment by kartoffelheinz** [comment]
This is a MAJOR pita and needs to be addressed as soon as possible. The app not respecting "only upload on non-metered wifi" not only cost me frequently money (remote location with sattelite uplink), it also causes distress in a team environment, where my device sucks up all the available bandwith for uploads which should only occur on non-metered wifis.

**Comment by garrett** [comment]
I've lost network connectivity for several days at the end of the month due to this bug. And I'm currently right before my cell phone plan's limit right now too. (My cell phone billing cycle resets at the first of the month. I may run out today or tomorrow.)

The problem has even gotten worse in the past year or so, as Google changed their Pixel camera app to have both raw and JPEGs in the same directory (so now, to have photos uploaded via Nextcloud, there's no way to have cloud sync but to upload both... but I don't want to upload the raw files unless I have WIFI... so my options are to not backup at all or risk losing all my cell phone data somewhere between halfway through the month to the end of the month, depending on how much I use my phone's camera). Even without this change, it was already a problem.

Either way, until this bug is fixed, it's really far from ideal; it's either:

1. lose complete access to my cell phone's data sometime in the month
2. pay penalties for running over (by extending the data plan for money), to have data again, and hope I don't have to keep paying
3. risk losing the photos I took (without a backup, if my phone breaks or is stolen)

...due to this specific bug.

It gets worse when traveling too (which usually means more photos, and may also even be considered roaming).

**Comment by agross** [comment]
@garrett I've since switched to [FolderSync](https://play.google.com/store/apps/details?id=dk.tacit.android.foldersync.lite&hl=en&gl=US) which allows you to be very specific about what to upload under which conditions. The devs here will not care enough given that it's been 4 years since I reported the bug. Certainly, the feature is not a core tenet of why people chose Nextcloud in the first place.

**Comment by wrenix** [comment]
Wow, stil a problem. I am surprised that nobody just patch this non functional option ("only upload on unmetered wifi")away. (Or fix it).

**Comment by agross** [comment]
Because I nobody cares. In the meantime I've switched to FolderSync for automatic uploads with actually working conditions and later to Immich for my photo needs.

**Comment by wrenix** [comment]
I have read your suggestion of a workaround, but sadly FolderSync is just in GooglePlay (not in F-Droid) and do not know if the source code is public.
